listenbrainz/webserver/views/api_tools.py

```diff
@@ -50,14 +52,8 @@ def insert_payload(payload, user, listen_type=LISTEN_TYPE_IMPORT):
     """ Convert the payload into augmented listens then submit them.
         Returns: augmented_listens
     """
-    try:
-        augmented_listens = _get_augmented_listens(payload, user)
-        _send_listens_to_queue(listen_type, augmented_listens)
-    except (APIInternalServerError, APIServiceUnavailable):
-        raise
-    except Exception as e:
-        current_app.logger.error("Error while inserting payload: %s", str(e), exc_info=True)
-        raise APIInternalServerError("Something went wrong. Please try again.")
+    augmented_listens = _get_augmented_listens(payload, user)
+    _send_listens_to_queue(listen_type, augmented_listens)
```

Comment:
The original version of this will give a generic error to the user, whereas the changed version will send the exact error from the exception. I know we talked about this and decided to remove the try/except, but I think that I'm in favour of hiding implementation details from the user where possible.
Maybe we should update our error handler to give a generic message (I see that [we already do this for API requests](https://github.com/metabrainz/listenbrainz-server/blob/ab61aae4481bd9f52a16301ea05617d4e963f0bf/listenbrainz/webserver/errors.py#L178-L182)). If we do this, will the original error still get caught properly by sentry?

Reply:
I'll test the sentry issue and see if we can give the user a generic message while retaining the full info in sentry.

Reply:
Tested and verified that the user is shown a generic error message while sentry does record the full error.